vendor/golang.org/x/net/http2/write.go

```diff
@@ -341,7 +341,12 @@ func encodeHeaders(enc *hpack.Encoder, h http.Header, keys []string) {
 	}
 	for _, k := range keys {
 		vv := h[k]
-		k = lowerHeader(k)
+		k, ascii := lowerHeader(k)
+		if !ascii {
+			// Skip writing invalid headers. Per RFC 7540, Section 8.1.2, header
+			// field names have to be ASCII characters (just as in HTTP/1.x).
+			continue
```

Comment:
this seems potentially notable.

Reply:
validWireHeaderFieldName below already skipped outputting strings which contained non-ascii characters ... this addition is catching things that were not ascii, but got transformed to ascii by strings.ToLower ... given they didn't work in http/1.x, and were being transformed to non-round-trippable lowercase headers, I'm not especially concerned about this

Reply:
I'm just not sure where all this is used, and it appears to be technically a breaking change, HTTP clients etc. seem notorious for doing / depending on out-of-spec stuff like this.